editor/src/components/canvas/canvas-strategies/shared-keyboard-strategy-helpers.ts

```diff
@@ -93,54 +93,58 @@ export function getKeyboardStrategyGuidelines(
   canvasState: InteractionCanvasState,
   interactionState: InteractionSession,
   draggedFrame: CanvasRectangle,
-): Array<GuidelineWithSnappingVector> {
+): Array<GuidelineWithSnappingVectorAndPointsOfRelevance> {
   const selectedElements = getTargetPathsFromInteractionTarget(canvasState.interactionTarget)
   const moveGuidelines = collectParentAndSiblingGuidelines(
     interactionState.metadata,
     selectedElements,
-  )
+  ).map((g) => g.guideline)
+
   const { horizontalPoints, verticalPoints } = Utils.getRectPointsAlongAxes(draggedFrame)
-  const closestGuideLines: Array<GuidelineWithSnappingVector> = mapDropNulls((guideline) => {
-    switch (guideline.type) {
-      case 'XAxisGuideline': {
-        const snappingVector = Guidelines.getOffsetToSnapToXGuideline(
-          horizontalPoints,
-          guideline,
-          null,
-        )
-        if (Utils.magnitude(snappingVector) === 0) {
-          return {
-            guideline: guideline,
-            snappingVector: snappingVector,
-            activateSnap: true,
+  const closestGuideLines: Array<GuidelineWithSnappingVectorAndPointsOfRelevance> = mapDropNulls(
+    (guideline) => {
+      switch (guideline.type) {
+        case 'XAxisGuideline': {
+          const snappingVector = Guidelines.getOffsetToSnapToXGuideline(
+            horizontalPoints,
+            guideline,
+            null,
+          )
+          if (Utils.magnitude(snappingVector) === 0) {
+            return {
+              guideline: guideline,
+              snappingVector: snappingVector,
+              pointsOfRelevance: [],
```

Comment:
From the PR alone I'm not clear on the previous intention of `activateSnap` here, and the result of replacing it with an empty `pointsOfReference` array 🤔 Is this changing the snapping behaviour here when using the keyboard?

Reply:
so I saw that `GuidelineWithSnappingVector` didn't include `activateSnap`, but this didn't cause a type error due to structural typing (since the object had everything `GuidelineWithSnappingVector` needed, plus `activateSnap`), so I removed `activateSnap` as cleanup

Reply:
keyboard snapping seems to be fine

Reply:
Aha, so my request to use a factory function is spot on then 😀